lib/zlib.js

```diff
@@ -507,8 +506,8 @@ function processChunkSync(self, chunk, flushFlag) {
       else
         buffers.push(out);
       nread += out.byteLength;
-    } else if (have < 0) {
-      assert(false, 'have should not go down');
+    } else {
+      assert(have >= 0, 'have should not go down');
```

Comment:
```suggestion
      assert(have === 0, 'have should not go down');
```

Reply:
It is not obvious from looking at the code why that check is sufficient. Do you have a small description why? :)